src/timeline.js

```diff
@@ -671,26 +715,38 @@ export default class Timeline extends React.Component {
             let endOffset = width > 0 ? left + width : left;
             const startTime = getTimeAtPixel(
               startOffset,
-              this.props.startDate,
-              this.props.endDate,
+              startDate,
+              endDate,
               this.getTimelineWidth(),
               this.getTimelineSnap()
             );
             const endTime = getTimeAtPixel(
               endOffset,
-              this.props.startDate,
-              this.props.endDate,
+              startDate,
+              endDate,
               this.getTimelineWidth(),
               this.getTimelineSnap()
             );
             //Get items in these ranges
             let selectedItems = [];
-            for (let r = Math.min(topRowNumber, bottomRow); r <= Math.max(topRowNumber, bottomRow); r++) {
-              selectedItems.push(
-                ..._.filter(this.rowItemMap[r], i => {
-                  return i.start.isBefore(endTime) && i.end.isAfter(startTime);
-                })
-              );
+
+            if (this.props.useTimestamps) {
+              let itemRows = _.groupBy(this.props.items, 'row');
```

Comment:
I was expecting a simpler code. I like `getDateAsMoment()`. And I was expecting some calls to this function. However, I see several places (this is not the only one), where "if useTimestamps" => a lot of logic. 1) How did you know that this has to be modified this way? 2) I'm affraid that there isn't a clear "recipe" for people that will contribute, in order to know what/if they need to do to respect this `useTimestamps`

I think the recipe that should be clearly documented in the class doc is that when accessing times from props or from row or from item: mandatory to pass through `getDateAsMoment()`. Forbidden to use it directly.

And applying this recipe, I searched ".start"; and I see a lot of places that respect this. But also some that don't. E.g. lines 406, 428, 454, maybe others. Shouldn't these also go through `getDateAsMoment()`?

I would have liked to have like in G4F, a function "getStartFromSegment()", instead of having "item.start". Maybe the user would like to store this in another field. And it would have allowed us to easily convert ts -> moment.

Reply:
Done.